claim/result_test.go

```diff
@@ -70,85 +72,196 @@ func TestResult_Validate(t *testing.T) {
+			outputName: map[string]string{
+				metadataKey: "old_value",
 			},
 		}
-		err := outputs.SetContentDigest("test1", contentDigest)
-		require.NoError(t, err, "SetContentDigest failed")
-		assert.Equal(t, wantoutputs, outputs, "SetContentDigest did not produce the expected structure")
+		err := outputs.SetMetadata(outputName, metadataKey, metadataValue)
+		require.NoError(t, err, "SetMetadata failed")
+		assert.Equal(t, wantoutputs, outputs, "SetMetadata did not produce the expected structure")
 	})
 
 	t.Run("existing invalid structure", func(t *testing.T) {
 		outputs := OutputMetadata{
-			"test1": map[string]interface{}{
-				OutputContentDigest: struct{}{},
+			outputName: map[string]interface{}{
+				metadataKey: struct{}{},
 			},
 		}
-		err := outputs.SetContentDigest("test1", contentDigest)
-		require.EqualError(t, err, "cannot set the claim result's OutputMetadata[test1][contentDigest] because it is not type map[string]string but map[string]interface {}")
+		err := outputs.SetMetadata(outputName, metadataKey, metadataValue)
+		require.EqualError(t, err, "cannot set the claim result's OutputMetadata[test1][so-meta] because it is not type map[string]string but map[string]interface {}")
 	})
 }
 
-func TestResultOutputs_GetContentDigest(t *testing.T) {
-	t.Run("output has digest", func(t *testing.T) {
-		contentDigest := "sha256:abc123"
+func TestResultOutputs_GetMetadata(t *testing.T) {
+	metadataKey := "so-meta"
+	outputName := "test1"
+	t.Run("output has metadata", func(t *testing.T) {
+		value := "that's so meta"
 
 		outputs := OutputMetadata{}
-		err := outputs.SetContentDigest("test1", contentDigest)
-		require.NoError(t, err, "SetContentDigest failed")
+		err := outputs.SetMetadata(outputName, metadataKey, value)
+		require.NoError(t, err, "SetMetadata failed")
 
-		gotContentDigest, ok := outputs.GetContentDigest("test1")
-		require.True(t, ok, "GetContentDigest should find the digest")
-		assert.Equal(t, contentDigest, gotContentDigest, "GetContentDigest should return the digest that we set")
+		gotValue, ok := outputs.GetMetadata(outputName, metadataKey)
+		require.True(t, ok, "GetMetadata should find the value")
+		assert.Equal(t, value, gotValue, "GetMetadata should return the value that we set")
 	})
 
 	t.Run("output not found", func(t *testing.T) {
 		outputs := OutputMetadata{}
-		gotContentDigest, ok := outputs.GetContentDigest("test1")
-		require.False(t, ok, "GetContentDigest should report that it did not find the contentDigest")
-		assert.Empty(t, gotContentDigest, "GetContentDigest should return an empty digest when one isn't found")
+		gotValue, ok := outputs.GetMetadata(outputName, metadataKey)
+		require.False(t, ok, "GetMetadata should report that it did not find the value")
+		assert.Empty(t, gotValue, "GetMetadata should return an empty value when one isn't found")
 	})
 
-	t.Run("output has no digest", func(t *testing.T) {
+	t.Run("output has no meatdata", func(t *testing.T) {
```

Comment:
I feel like this is an easter egg for PR reviewers 🥩  and I almost don't want to change it buuut... `s/meatdata/metadata`?

Reply:
I am pro meatdata